test/unit/services/tracker.ts

```diff
@@ -75,6 +71,60 @@ suite('Tracker Service', ({ expect, spy, stub, itShouldExtendBaseService }) => {
     });
   });
 
+  describe('setListeners()', () => {
+    it('should listen for events', () => {
+      const applyOptsOverride = spy(service, 'applyOptsOverride');
+      const sendSearchEvent = service.sendSearchEvent = spy();
+      const sendViewCartEvent = service.sendViewCartEvent = spy();
+      const sendAddToCartEvent = service.sendAddToCartEvent = spy();
+      const sendRemoveFromCartEvent = service.sendRemoveFromCartEvent = spy();
+      const sendOrderEvent = service.sendOrderEvent = spy();
+      const sendViewProductEvent = service.sendViewProductEvent = spy();
+      const sendMoreRefinementsEvent = service.sendMoreRefinementsEvent = spy();
+      const match = (fn, override) => sinon.match((value) => {
+        value();
+        expect(applyOptsOverride).to.be.calledWith(fn, override);
+        expect(fn).to.be.called;
+        return true;
+      });
+      on = app.flux.on = spy();
+
+      service.setListeners(app);
+
+      expect(on).to.have.callCount(7);
+      // tslint:disable max-line-length
+      expect(on).to.be.calledWith(Events.BEACON_SEARCH, match(sendSearchEvent, service['opts'].sendSearchEvent));
```

Comment:
Is there a way we can change the types so we don't have to use `[]` to access `opts`?